src/category_theory/enriched/basic.lean

```diff
@@ -0,0 +1,292 @@
+-/
+@[ext, nolint has_inhabited_instance]
+structure graded_nat_trans (A : V) (F G : enriched_functor V C D) :=
+(app : Π (X : C), A ⟶ (F.obj X ⟶[V] G.obj X))
+(naturality :
+  ∀ (X Y : C), (app Y ⊗ F.map X Y) ≫ (β_ _ _).hom ≫ e_comp V _ _ _ =
+    (app X ⊗ G.map X Y) ≫ e_comp V _ _ _)
+
+/--
+A presheaf isomorphic to the Yoneda embedding of
+the `V`-object of natural transformations from `F` to `G`.
+-/
+@[simps]
+def enriched_nat_trans_yoneda (F G : enriched_functor V C D) : Vᵒᵖ ⥤ (Type (max u₁ w)) :=
+{ obj := λ A, graded_nat_trans (unop A) F G,
+  map := λ A A' f σ,
+  { app := λ X, f.unop ≫ σ.app X,
+    naturality := λ X Y, begin
+      rw [←tensor_id_comp_id_tensor _ (f.unop ≫ σ.app Y),
+        ←tensor_id_comp_id_tensor _ (f.unop ≫ σ.app X),
+        comp_tensor_id, comp_tensor_id,
+        category.assoc, category.assoc, category.assoc, category.assoc,
+        tensor_id_comp_id_tensor_assoc, tensor_id_comp_id_tensor_assoc,
+        σ.naturality],
+     end }, }
+
+-- TODO assuming `[has_limits C]` construct the actual object of natural transformations
+-- and show that the functor category is `V`-enriched.
+
+end
+
+/--
+Construct an honest category from a `Type v`-enriched category.
+-/
+def category_of_enriched_category_Type (C : Type u₁) [𝒞 : enriched_category (Type v) C] :
+  category.{v} C :=
+{ hom := 𝒞.hom,
+  id := λ X, e_id (Type v) X punit.star,
+  comp := λ X Y Z f g, e_comp (Type v) X Y Z ⟨f, g⟩,
+  id_comp' := λ X Y f, congr_fun (e_id_comp (Type v) X Y) f,
+  comp_id' := λ X Y f, congr_fun (e_comp_id (Type v) X Y) f,
+  assoc' := λ W X Y Z f g h, (congr_fun (e_assoc (Type v) W X Y Z) ⟨f, g, h⟩ : _), }
+
+/--
+Construct a `Type v`-enriched category from an honest category.
+-/
+def enriched_category_Type_of_category (C : Type u₁) [𝒞 : category.{v} C] :
+  enriched_category (Type v) C :=
+{ hom := 𝒞.hom,
+  id := λ X p, 𝟙 X,
+  comp := λ X Y Z p, p.1 ≫ p.2,
+  id_comp := λ X Y, by { ext, simp, },
+  comp_id := λ X Y, by { ext, simp, },
+  assoc := λ W X Y Z, by { ext ⟨f, g, h⟩, simp, }, }
+
+/--
+We verify that an enriched category in `Type u` is just the same thing as an honest category.
+-/
+def enriched_category_Type_equiv_category (C : Type u₁) :
+  (enriched_category (Type v) C) ≃ category.{v} C :=
+{ to_fun := λ 𝒞, by exactI category_of_enriched_category_Type C,
+  inv_fun := λ 𝒞, by exactI enriched_category_Type_of_category C,
+  left_inv := λ 𝒞, begin
+    cases 𝒞,
+    dsimp [enriched_category_Type_of_category],
+    congr,
+    { ext X ⟨⟩, refl, },
+    { ext X Y Z ⟨f, g⟩, refl, }
+  end,
+  right_inv := λ 𝒞, by { rcases 𝒞 with ⟨⟨⟨⟩⟩⟩, dsimp, congr, }, }.
+
+section
+local attribute [instance] category_of_enriched_category_Type
+
+/--
+We verify that an enriched functor between `Type v` enriched categories
+is just the same thing as an honest functor.
+-/
+@[simps]
+def enriched_functor_Type_equiv_functor
+  {C : Type u₁} [𝒞 : enriched_category (Type v) C]
+  {D : Type u₂} [𝒟 : enriched_category (Type v) D] :
```

Comment:
The explicit names aren't necessary here, right?

Reply:
Fixed.